erpnext/stock/report/stock_balance_multi_uom/stock_balance_multi_uom.py

```diff
@@ -0,0 +1,299 @@
+		for key, val in qty_dict.items():
+			val = flt(val, float_precision)
+			qty_dict[key] = val
+			if key != "val_rate" and val:
+				no_transactions = False
+
+		if no_transactions:
+			iwb_map.pop((company, item, warehouse))
+
+	return iwb_map
+
+def get_items(filters):
+	conditions = []
+	if filters.get("item_code"):
+		conditions.append("item.name=%(item_code)s")
+	else:
+		if filters.get("brand"):
+			conditions.append("item.brand=%(brand)s")
+		if filters.get("item_group"):
+			conditions.append(get_item_group_condition(filters.get("item_group")))
+
+	items = []
+	if conditions:
+		items = frappe.db.sql_list("""select name from `tabItem` item where {}"""
+			.format(" and ".join(conditions)), filters)
+	return items
+
+def get_item_details(items, sle, filters):
+	item_details = {}
+	if not items:
+		items = list(set([d.item_code for d in sle]))
+
+	if items:
+		for item in frappe.db.sql("""
+			select name, item_name, description, item_group, brand, stock_uom ,  purchase_uom ,
+			(SELECT uomc.conversion_factor
+              FROM `tabUOM Conversion Detail` uomc
+             WHERE uomc.parent = itm.name
+                AND uomc.uom = purchase_uom ) purchase_uom_conversion_factor,
+			 sales_uom ,
+			 (SELECT uomc.conversion_factor
+               FROM `tabUOM Conversion Detail` uomc
+              WHERE uomc.parent = itm.name
+                AND uomc.uom = sales_uom ) sales_uom_conversion_factor
+			from `tabItem` itm
+			where itm.name in ({0}) and ifnull(disabled, 0) = 0
+			""".format(', '.join(['"' + frappe.db.escape(i, percent=False) + '"' for i in items])), as_dict=1):
+				item_details.setdefault(item.name, item)
+
+	if filters.get('show_variant_attributes', 0) == 1:
+		variant_values = get_variant_values_for(item_details.keys())
+		item_details = {k: v.update(variant_values.get(k, {})) for k, v in item_details.iteritems()}
+
+	return item_details
+
+def get_item_reorder_details(items):
+	item_reorder_details = frappe._dict()
+
+	if items:
+		item_reorder_details = frappe.db.sql("""
+			select parent, warehouse, warehouse_reorder_qty, warehouse_reorder_level
+			from `tabItem Reorder`
+			where parent in ({0})
+		""".format(', '.join(['"' + frappe.db.escape(i, percent=False) + '"' for i in items])), as_dict=1)
+
+	return dict((d.parent + d.warehouse, d) for d in item_reorder_details)
+
+def validate_filters(filters):
+	if not (filters.get("item_code") or filters.get("warehouse")):
+		sle_count = flt(frappe.db.sql("""select count(name) from `tabStock Ledger Entry`""")[0][0])
+		if sle_count > 500000:
+			frappe.throw(_("Please set filter based on Item or Warehouse"))
+
+def get_variants_attributes():
+	'''Return all item variant attributes.'''
+	return [i.name for i in frappe.get_all('Item Attribute')]
+
+def get_variant_values_for(items):
+	'''Returns variant values for items.'''
+	attribute_map = {}
+	for attr in frappe.db.sql('''select parent, attribute, attribute_value
+		from `tabItem Variant Attribute` where parent in (%s)
```

Comment:
Please follow the indentation guidelines:

https://github.com/frappe/erpnext/wiki/Coding-Standards#use-simple-indenting